src/Aseo/Api/Auth/KeyAuth.php

```diff
@@ -127,6 +132,10 @@ public function setApiKey($apiKey)
      */
     public function getTimestamp()
     {
+        if (null === $this->timestamp) {
```

Comment:
This seems to be a fairly common practice of assigning default values that can not be hardcoded

throwing an exception means that for each api call our customers would have to instantiate a new Auth class, that seems a bit excessive to me

calling the setter method, sure, that strikes me as a good way to ensure there we always use the setter to manipulate properties